$MirOS: ports/x11/gnome/vfs2/patches/patch-libgnomevfs-gnome-vfs-dns-sd_c,v 1.2 2005/10/15 17:23:49 bsiegert Exp $
--- libgnomevfs/gnome-vfs-dns-sd.c.orig	Tue Apr 10 14:03:36 2007
+++ libgnomevfs/gnome-vfs-dns-sd.c	Sun Jun 17 13:38:05 2007
@@ -85,7 +85,7 @@ typedef struct {
 } dns_message_header;
 
 typedef struct {
-	char name[NS_MAXDNAME];
+       char name[MAXDNAME];
 	guint16 type;
 	guint16 class;
 	guint32 ttl;
@@ -250,7 +250,7 @@ parse_header (unsigned char *reply, int 
 	      unsigned char *p,
 	      dns_message_header *header)
 {
-	if (reply_len < NS_HFIXEDSZ)
+       if (reply_len < HFIXEDSZ)
 		return -1;
 
 	header->id = decode_16 (p); p += 2;
@@ -260,7 +260,7 @@ parse_header (unsigned char *reply, int 
 	header->nscount = decode_16 (p); p += 2;
 	header->arcount = decode_16 (p); p += 2;
 
-	return NS_HFIXEDSZ;
+	return HFIXEDSZ;
 }
 
 
@@ -333,7 +333,7 @@ unicast_list_domains_sync (const char *d
 	unsigned char reply[DNS_REPLY_SIZE];
 	unsigned char *p;
 	char *searchdomain;
-	char name[NS_MAXDNAME];
+	char name[MAXDNAME];
 	dns_message_header header;
 	dns_message_rr rr;
 	int res;
@@ -349,7 +349,7 @@ unicast_list_domains_sync (const char *d
 	/* Use TCP to support large queries */
 	_res.options |= RES_USEVC;
 	searchdomain = g_strconcat ("_browse._dns-sd._udp.", domain, NULL);
-	reply_len = res_search (searchdomain, ns_c_in, ns_t_ptr,
+	reply_len = res_search (searchdomain, C_IN, T_PTR,
 				reply, sizeof (reply));
 	g_free (searchdomain);
 	if (reply_len == -1) {
@@ -393,7 +393,7 @@ unicast_list_domains_sync (const char *d
 			goto error;
 		p += len;
 		
-		if (rr.type == ns_t_ptr) {
+		if (rr.type == T_PTR) {
 			len = dn_expand (reply, reply + reply_len, p, name, sizeof(name));
 			if (len < 0) 
 				goto error;
@@ -444,7 +444,7 @@ unicast_browse_sync (const char *domain,
 	unsigned char reply[DNS_REPLY_SIZE];
 	unsigned char *p;
 	char *searchdomain;
-	char name[NS_MAXDNAME];
+	char name[MAXDNAME];
 	dns_message_header header;
 	dns_message_rr rr;
 	GArray *array;
@@ -461,7 +461,7 @@ unicast_browse_sync (const char *domain,
 	/* Use TCP to support large queries */
 	_res.options |= RES_USEVC;
 	searchdomain = g_strconcat (type, ".", domain, NULL);
-	reply_len = res_search (searchdomain, ns_c_in, ns_t_ptr,
+	reply_len = res_search (searchdomain, C_IN, T_PTR,
 				reply, sizeof (reply));
 	g_free (searchdomain);
 	if (reply_len == -1) {
@@ -507,11 +507,12 @@ unicast_browse_sync (const char *domain,
 			goto error;
 		p += len;
 		
-		if (rr.type == ns_t_ptr) {
+		if (rr.type == T_PTR) {
 			GnomeVFSDNSSDService service;
-			char ptr_service[NS_MAXDNAME];
-			char ptr_type[NS_MAXDNAME];
-			char ptr_domain[NS_MAXDNAME];
+
+			char ptr_service[MAXDNAME];
+			char ptr_type[MAXDNAME];
+			char ptr_domain[MAXDNAME];
 			
 			len = dn_expand (reply, reply + reply_len, p, name, sizeof(name));
 			if (len < 0) 
@@ -605,7 +606,7 @@ unicast_resolve_sync (const char *name,
 	int reply_len, len, i;
 	unsigned char reply[DNS_REPLY_SIZE];
 	unsigned char *p;
-	char dnsname[NS_MAXDNAME];
+	char dnsname[MAXDNAME];
 	dns_message_header header;
 	dns_message_rr rr;
 	char *full_name;
@@ -625,7 +626,7 @@ unicast_resolve_sync (const char *name,
 	_res.options |= RES_USEVC;
   
 	full_name = service_to_dns_name (name, type, domain);
-	reply_len = res_search (full_name, ns_c_in, ns_t_any,
+	reply_len = res_search (full_name, C_IN, T_ANY,
 				reply, sizeof (reply));
 	g_free (full_name);
 	if (reply_len == -1) {
@@ -669,7 +670,7 @@ unicast_resolve_sync (const char *name,
 			goto error;
 		p += len;
 		
-		if (rr.type == ns_t_srv) {
+		if (rr.type == T_SRV) {
 			unsigned char *pp;
 			int priority, weight, port;
 			
@@ -690,7 +691,7 @@ unicast_resolve_sync (const char *name,
 			}
 		}
 		
-		if (rr.type == ns_t_txt) {
+		if (rr.type == T_TXT) {
 			*text_raw_out = g_memdup (p, rr.rdlength);
 			*text_raw_len_out = rr.rdlength;
 		}
